Can't build target release on windows

<!-- Please search existing issues for potential duplicates before filing yours:
https://github.com/godotengine/godot/issues?q=is%3Aissue
-->

**Godot version:**
5614692a1


**OS/device including version:**
Windows 10


**Issue description:**
Building with:
```
scons -j12 platform=windows target=release tools=no
```
yields:
```
[Initial build] Compiling ==> modules\gridmap\grid_map.cpp
grid_map.cpp
modules\gdscript\gdscript_parser.cpp(4446): error C2027: wykorzystanie niezdefiniowanego przez użytkownika typu "GDScriptWarning"
d:\projects\godot\godot-src\modules\gdscript\gdscript_parser.h(41): note: zobacz deklarację „GDScriptWarning”
modules\gdscript\gdscript_parser.cpp(4446): error C2065: "DEPRECATED_KEYWORD": niezadeklarowany identyfikator
modules\gdscript\gdscript_parser.cpp(4446): error C3861: "_add_warning": identyfikatora nie odnaleziono
scons: *** [modules\gdscript\gdscript_parser.windows.opt.64.obj] Error 2
scons: building terminated because of errors.
```
With either `tools=yes` or `target=debug` build works fine.

**Steps to reproduce:**
```
scons -j12 platform=windows target=release tools=no
```

--- a/modules/gdscript/gdscript_parser.cpp
+++ b/modules/gdscript/gdscript_parser.cpp
@@ -4443,7 +4443,9 @@ void GDScriptParser::_parse_class(ClassNode *p_class) {
 				continue;
 			} break;
 			case GDScriptTokenizer::TK_PR_SLAVE:
+#ifdef DEBUG_ENABLED
 				_add_warning(GDScriptWarning::DEPRECATED_KEYWORD, tokenizer->get_token_line(), "slave", "puppet");
+#endif
 			case GDScriptTokenizer::TK_PR_PUPPET: {
 
 				//may be fallthrough from export, ignore if so

